server/conn.go

```diff
@@ -38,18 +33,18 @@ type conn struct {
 	conn net.Conn
 }
 
-func newConn(s *Server, netConn net.Conn) (*conn, error) {
+func newConn(s *Server, netConn net.Conn, bufrw *bufio.ReadWriter) (*conn, error) {
 	s.connsLock.Lock()
 	defer s.connsLock.Unlock()
 
 	if !s.isLeader() {
-		return nil, errors.Errorf("server <%s> is not leader, cannot create new connection <%s>", s.cfg.AdvertiseAddr, netConn.RemoteAddr())
+		return nil, errors.Errorf("server <%s> is not leader, cannot create new connection <%s>", s.Name(), netConn.RemoteAddr())
 	}
 
 	c := &conn{
 		s:    s,
-		rb:   bufio.NewReaderSize(netConn, readBufferSize),
-		wb:   bufio.NewWriterSize(netConn, writeBufferSize),
+		rb:   bufrw.Reader,
```

Comment:
Seems we don't have any better solutions yet, I just use the original `bufrw` here.